openai/loaders/vision.ts

```diff
@@ -0,0 +1,89 @@
+import puppeteer from "https://deno.land/x/puppeteer@16.2.0/mod.ts";
+import { AppContext } from "../mod.ts";
+
+export interface Props {
+  /**
+   * @description the links of the site that contains images
+   * @examples "https://www.instagram.com/marcoscandeia"
+   */
+  url: string;
+  /**
+   * @description what kind of description do you want?
+   * @examples you can ask for something like "What’s in this image?"\n or "Is there a car?"
+   */
+  request?: string;
+  /**
+   * @description
+   * @examples 5
+   * @default 5
+   */
+  imageCount?: number;
+}
+
+const getImagesFrom = async (url: string): Promise<string[]> => {
+  const browser = await puppeteer.launch({
+    args: ["--no-sandbox"],
+    headless: true,
+  });
+  const page = await browser.newPage();
+  await page.setUserAgent(
+    "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.169 Safari/537.36",
+  );
+  const images: string[] = [];
+  page.on("response", (response) => {
+    const headers = response.headers();
+    const contentType = headers["Content-Type"] ?? headers["content-type"];
+    if (contentType?.includes("jpeg")) {
+      images.push(response.url());
+    }
+  });
+  await page.goto(url);
+  await page.waitForNetworkIdle({ idleTime: 3000 });
+  await browser.close();
+  return images;
+};
```

Comment:
remove puppeteer dependency.

let vision loader receive an `ImageSource` property that can be provided by a puppeteer img loader but that can also be just a fetch on an image.

Reply:
Im going to delete this loader since we're planning to not use instagram